features/ui/button/button.tsx

```diff
@@ -18,3 +54,184 @@ export const Button = styled.button`
+        return css`
+          background-color: ${color("primary", 600)};
+          &:hover {
+            background-color: ${color("primary", 700)};
+          }
+          &:focus {
+            outline: 3px solid ${color("primary", 100)};
+          }
+          &:disabled {
+            background-color: ${color("primary", 200)};
+          }
+        `;
+      case ButtonColor.secondary:
+        return css`
+          color: ${color("primary", 700)};
+          background-color: ${color("primary", 50)};
+          &:hover {
+            background-color: ${color("primary", 100)};
+          }
+          &:focus {
+            outline: 3px solid ${color("primary", 100)};
+            background-color: ${color("primary", 50)};
+          }
+          &:disabled {
+            background-color: ${color("primary", 25)};
+            color: ${color("primary", 300)};
+          }
+        `;
+      case ButtonColor.gray:
+        return css`
+          color: ${color("gray", 700)};
+          border: 1px solid ${color("gray", 300)};
+          background-color: rgba(255, 255, 255, 1);
+          &:hover {
+            color: ${color("gray", 800)};
+            background-color: ${color("gray", 50)};
+          }
+          &:focus {
+            outline: 3px solid ${color("gray", 100)};
+            background-color: rgba(255, 255, 255, 1);
+          }
+          &:disabled {
+            background-color: ${color("gray", 200)};
+            color: ${color("gray", 300)};
+          }
+        `;
+      case ButtonColor.empty:
+        return css`
+          color: ${color("primary", 700)};
+          border: none;
+          background-color: rgba(0, 0, 0, 0);
+          &:hover {
+            background-color: ${color("primary", 50)};
+          }
+          &:focus {
+            outline: none;
+            background-color: rgba(255, 255, 255, 1);
+          }
+          &:disabled {
+            color: ${color("gray", 300)};
+          }
+        `;
+      case ButtonColor.emptyGray:
+        return css`
+          color: ${color("gray", 500)};
+          border: none;
+          background-color: rgba(0, 0, 0, 0);
+          &:hover {
+            background-color: ${color("gray", 50)};
+            color: ${color("gray", 600)};
+          }
+          &:focus {
+            outline: none;
+            background-color: rgba(0, 0, 0, 0);
+          }
+          &:disabled {
+            color: ${color("gray", 300)};
+          }
+        `;
+      case ButtonColor.error:
+        return css`
+          color: #ffffff;
+          border: 1px;
+          background-color: ${color("error", 600)};
+          &:hover {
+            background-color: ${color("error", 700)};
+          }
+          &:focus {
+            outline: 3px solid ${color("error", 100)};
+          }
+          &:disabled {
+            background-color: ${color("error", 200)};
+          }
+        `;
+    }
+  }}
+`;
+
+export type ButtonContainerProps = {
```

Comment:
For components in UI libraries it's often best to use unions of custom prop types with existing HTML element types. Here for example you can write

```
export type ButtonContainerProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  size?: ButtonSize;
  color?: ButtonColor;
  icon?: ButtonIcon;
};
```

The `children` and `disabled` props are already defined on the `ButtonHTMLAttributes<HTMLButtonElement>` type. And here for example you missed the `onClick` prop. 

Then you can define the component like this:

```
export const ButtonContainer = ({
  children,
  size = ButtonSize.md,
  color = ButtonColor.primary,
  icon = ButtonIcon.none,
  ...buttonProps
}: ButtonContainerProps) => {
  return (
    <Container size={size} color={color} icon={icon} {...buttonProps}>
```